README.md

```diff
@@ -16,15 +17,138 @@ TerseTS can be compiled and cross-compiled from source:
+    
+   var decompressed_values = try tersets.decompress(compressed, config);
+   defer std.heap.page_allocator.free(decompressed);
+
+   std.debug.print("Decompression successful. Decompressed data length {any}\n", .{decompressed_values.items.len});
+}
+```
+
+TerseTS provides `./src/tersets.zig` as the single access point and two main functions `compress` and `decompress`. 
+
+For compression, you can select the compression method through the `Configuration` structure with two parameters: the compression method, e.g., `.method=.SwingFilter`, and the error bound, e.g., `.error_bound = 0.1`. The supported compression methods are specified in `src/tersets.zig`. 
+
+For decompression, the `Configuration` is not needed as the method is encoded in the compressed values.
+</details>
+
+<a id="c-usage-example"></a>
+<details>
+<summary><strong>C/C++ Usage Example</strong></summary>
+
+```c 
+#include "tersets.h"
+#include <stdio.h>
+
+int main() {
+   double uncompressed_values[] = {1.0, 2.0, 3.0, 4.0, 5.0};
+   struct UncompressedValues uncompressed_values = {data, 5};
+   
+   printf("Uncompressed data length: %lu\n", uncompressed_values.len);
+   
+   // Configuration for compression. 
+   // The supported compression methods are specified in tersets.zig.
+   // Method 2 is SwingFilter and 0.1 error bound.
+   struct Configuration config = {2, 0.0}; 
+
+   // Prepare for compressed data. 
+   // The compressed values point to dynamically allocated data that should be deallocated.
+   struct CompressedValues compressed_values;
+    
+   // Compress the data.
+   int32_t result = compress(uncompressed_values, &compressed_values, config);
+   if (result != 0) {
+      printf("Compression failed with error code %d\n", result);
+      return -1;
+   }
+
+   printf("Compression successful. Decompressed data length: %lu\n", compressed_values.len);
+    
+   // Prepare for decompressed data. 
+   // The decompressed values point to dynamically allocated data that should be deallocated.
+   struct UncompressedValues decompressed_values;
+   
+   // Decompress the data.
+   int32_t result = decompress(compressed_values, &decompressed_values);
+   if (result != 0) {
+      printf("Decompression failed with error code %d\n", result);
+      return -1;
+   }
+
+   printf("Decompression successful. Decompressed data length: %lu\n", decompressed_values.len);
+    
+   // Free the compressed and decompressed value.
+   free(decompressed_values.data);
+   free(compressed_values.data);
+   return 0;
+}
+```
+
+TerseTS provides `./bindings/c/tersets.h` as API for C/C++ which should be included in the source code, i.e., `#include "tersets.h"`. You need to [link](#linking) the TerseTS library to the project. 
+
+Ensure that the method field in `Configuration` is set to a valid compression/decompression method supported by `TerseTS`.
+
+Free dynamically allocated memory appropriately to avoid memory leaks.
```

Comment:
Seems something is missing from this sentence, maybe start the sentence with "Remember to free dynamically....`

Reply:
Fixed.